File: java/src/test/java/com/example/demo/ApplicationTests.java
package com.example.demo;

import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.http.MediaType;
import org.springframework.test.web.servlet.MockMvc;
import org.springframework.test.web.servlet.setup.MockMvcBuilders;
import org.springframework.web.context.WebApplicationContext;

import static org.hamcrest.CoreMatchers.containsString;
import static org.hamcrest.CoreMatchers.equalTo;
import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;

@SpringBootTest
public class ApplicationTests {


    @Autowired
    WebApplicationContext applicationContext;

    private MockMvc mvc;

    @BeforeAll
    public void setup() {
        this.mvc = MockMvcBuilders.webAppContextSetup(applicationContext).build();
    }

    @Test
    @DisplayName("get blog properties")
    public void getBlogProperties() throws Exception {
        this.mvc
                .perform(
                        get("/info")
                                .accept(MediaType.APPLICATION_JSON)
                )
                .andExpect(status().isOk())
                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
                .andExpect(jsonPath("$.title", equalTo("Hantsy'Blog")))
                .andExpect(jsonPath("$.description", equalTo("Description of Hantsy's Blog")))
                .andExpect(jsonPath("$.author", equalTo("Hantsy")));
    }

    @Test
    @DisplayName("get all posts should be ok")
    public void getPosts_shouldOK() throws Exception {
        this.mvc
                .perform(
                        get("/posts")
                                .accept(MediaType.APPLICATION_JSON)
                )
                .andExpect(status().isOk())
                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
                .andExpect(jsonPath("$.[0].title", containsString("one")))
                .andExpect(jsonPath("$.[1].title", containsString("two")));
    }

}

File: kotlin/src/test/kotlin/com/example/demo/ApplicationTests.kt
package com.example.demo


import org.hamcrest.Matchers.`is`
import org.hamcrest.Matchers.containsString
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.context.SpringBootTest.WebEnvironment.MOCK
import org.springframework.context.ApplicationContextInitializer
import org.springframework.context.support.GenericApplicationContext
import org.springframework.http.MediaType.APPLICATION_JSON
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.get
import org.springframework.test.web.servlet.setup.MockMvcBuilders.webAppContextSetup
import org.springframework.web.context.WebApplicationContext
import java.util.*


class TestConfigInitializer : ApplicationContextInitializer<GenericApplicationContext> {
    override fun initialize(applicationContext: GenericApplicationContext) {
        beans.initialize(applicationContext)
    }

}

@SpringBootTest(
        webEnvironment = MOCK,
        properties = ["context.initializer.classes=com.example.demo.TestConfigInitializer"]
)
class DemoApplicationTests {

    private lateinit var mockMvc: MockMvc

    @Autowired
    private lateinit var appContext: WebApplicationContext


    @BeforeEach
    fun setup() {
        mockMvc = webAppContextSetup(appContext).build();
    }

    @Test
    fun `Get blog properties`() {
        mockMvc
                .get("/info") {
                    accept = APPLICATION_JSON
                }
                .andExpect {
                    status { isOk }
                    content { contentType(APPLICATION_JSON) }
                    jsonPath("$.title") { value(`is`("Hantsy'Blog"))}
                    jsonPath("$.description") { value(`is`("Description of Hantsy's Blog"))}
                    jsonPath("$.author") { value(`is`("Hantsy"))} //content { json("""{"someBoolean": false}""", false) }
                }
                .andDo {
                    print()
                }
    }

    @Test
    fun `Get all posts should ok`() {
        mockMvc
                .get("/posts") {
                    //secure = true
                    accept = APPLICATION_JSON
                    headers {
                        contentLanguage = Locale.ENGLISH
                    }
                    //principal = Principal { "foo" }
                }
                .andExpect {
                    status { isOk }
                    content { contentType(APPLICATION_JSON) }
                    jsonPath("$[0].title") { value(containsString("post")) } //content { json("""{"someBoolean": false}""", false) }
                }
                .andDo {
                    print()
                }
    }

}
